bin/compile

```diff
@@ -13,6 +13,22 @@ source "${buildpack}/lib/common.sh"
 source "${buildpack}/lib/common_tools.sh"
 source_stdlib
 
+resetCache() {
+    rm -rf "${cache:?}"/*
+    echo "$STACK" > "$cache/stack"
```

Comment:
What are the trickle-down effects if $STACK isn't set?

Reply:
What are the effects of stacking this buildpack with other buildpacks that "reset the stack" ?

Reply:
Changed this up a little. Before, the stack value was only being cached if it changed. Now, it's always cached. If STACK is unset or empty, then an empty value is cached. I think that's okay, though, since changes will still be detected.

Re composing this buildpack with others, I also had this one cache the stack in `$cache/go-stack` instead of the more generic `$cache/stack`. I'm not sure what the expectations are around how buildpacks compose with each other, who owns the cache, etc. I think the changes I mentioned above will work well if another buildpack nukes the entire cache as this one does and has been.

It might be nice at some point to make an effort to better namespace this buildpack's use of the cache so it's easier to only clear its part, like putting everything under `$cache/go` or something.